ui-spacetimechart/src/utils/png.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * This function captures a list of canvas layers into a PNG image, and returns a blob with the
+ * PNG data if possible.
+ */
+export default async function getPNGBlob(
+  canvases: Record<string, HTMLCanvasElement>,
+  layers: readonly string[]
+): Promise<Blob> {
+  if (!layers.length) throw Error('There must be at least one layer to capture.');
+
+  const firstCanvas = canvases[layers[0]];
+  const width = firstCanvas.width;
+  const height = firstCanvas.height;
+
+  // Create a new canvas, on which the different layers will be drawn:
+  const canvas = document.createElement('CANVAS') as HTMLCanvasElement;
+  canvas.setAttribute('width', width + '');
+  canvas.setAttribute('height', height + '');
```

Comment:
I personnally don't agree with that one. `String`, as well as `Boolean` and `Number`, are very misleading in JavaScript, because they are class constructors, initially designed to handle mutable objects-like reference-based strings, booleans and numbers - while we almost always handle those values as immutable value-based strings, booleans and numbers. The fact that we can use them as casting functions is weird, in terms of design.

I fear that some JavaScript beginner, seeing `const w = String(width);` might think is it a typo and should be `const w = new String(width);` instead, while `const w = width + '';` is unique and enough commonly used. I have the same opinion on `const isEmpty = !!array.length;` and `const now = +(new Date());`.

Now, if you want to enforce the use of these types of casting, we can add the [`no-implicit-coercion` ESLint rule](https://eslint.org/docs/latest/rules/no-implicit-coercion).

Reply:
You can also do ``` `${width}` ``` ?

Reply:
Yes, it does the exact same as `width + ''` I think.